### PR TITLE
TIP-856: Move Pim\AkeneoStorageUtils Component and Bundle to Akeneo\Tool

### PR DESCRIPTION
**Pull request by arnolanglade**
**Description (for Contributor and Core Developer)**

A new name space has been created in this PR Akeneo\Tool which will contains our tools. All components and bundles from Akeneo\Xxx will be move to Akeneo\Tool\Xxx.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-3.0.md [hunk 1]
```diff
@@ -53,8 +53,10 @@
...
-- Move namespace  `Pim\Bundle\ApiBundle` to `Akeneo\Tool\Bundle\ApiBundle`
+- Move namespace `Pim\Bundle\ApiBundle` to `Akeneo\Tool\Bundle\ApiBundle`
 - Move namespace `Pim\Component\Batch` to `Akeneo\Tool\Component\Batch`
-- Move namespace  `Pim\Bundle\BatchBundle` to `Akeneo\Tool\Bundle\BatchBundle`
+- Move namespace `Pim\Bundle\BatchBundle` to `Akeneo\Tool\Bundle\BatchBundle`
 - Move namespace `Pim\Component\BatchQueue` to `Akeneo\Tool\Component\BatchQueue`
-- Move namespace  `Pim\Bundle\BatchQueueBundle` to `Akeneo\Tool\Bundle\BatchQueueBundle`
+- Move namespace `Pim\Bundle\BatchQueueBundle` to `Akeneo\Tool\Bundle\BatchQueueBundle`
+- Move namespace `Pim\Component\StorageUtilsQueue` to `Akeneo\Tool\Component\StorageUtilsQueue`
+- Move namespace `Pim\Bundle\StorageUtilsQueueBundle` to `Akeneo\Tool\Bundle\StorageUtilsQueueBundle`
```
Comment (pierallard) [inline CHANGELOG-3.0.md]: Thanks :dancing_women:

File: src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeSearchableRepository.php [hunk 2]
```diff
@@ -105,7 +105,7 @@ protected function resolveOptions(array $options)
     protected function findBySearchQb($search, array $options)
     {
         //TODO: refactor on master because this is exactly the same that FamilySearchableRepository
-        //TODO: and should be put in Akeneo\Bundle\StorageUtilsBundle\Doctrine\ORM\Repository\SearchableRepository
+        //TODO: and should be put in Akeneo\Tool\Bundle\StorageUtilsBundle\Doctrine\ORM\Repository\SearchableRepository
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeSearchableRepository.php]: Is it the opportunity to do this too in this PR ?

Comment (jjanvier) [reply src/Pim/Bundle/EnrichBundle/Doctrine/ORM/Repository/AttributeSearchableRepository.php]: nope ^^
no refactor in our PRs
we just move stuff at the moment

File: src/Akeneo/Tool/Component/StorageUtils/spec/Cursor/PaginatorFactorySpec.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+namespace spec\Akeneo\Tool\Component\StorageUtils\Cursor;
+
+use Akeneo\Tool\Component\StorageUtils\Cursor\CursorInterface;
+use Akeneo\Tool\Component\StorageUtils\Cursor\Paginator;
+use Akeneo\Tool\Component\StorageUtils\Cursor\PaginatorFactory;
+use Akeneo\Tool\Component\StorageUtils\Cursor\PaginatorFactoryInterface;
+use Akeneo\Tool\Component\StorageUtils\Cursor\PaginatorInterface;
+use PhpSpec\ObjectBehavior;
+
+class PaginatorFactorySpec extends ObjectBehavior
```
Comment (jjanvier) [inline src/Akeneo/Tool/Component/StorageUtils/spec/Cursor/PaginatorFactorySpec.php]: did you add the spec or is it a bad move?

Comment (arnolanglade) [reply src/Akeneo/Tool/Component/StorageUtils/spec/Cursor/PaginatorFactorySpec.php]: it seems that phpstorm did not move this one.

**Pull request merged**